wandb/__init__.py

```diff
@@ -483,6 +511,17 @@ def init(job_type=None, dir=None, config=None, project=None, entity=None, group=
                            "WANDB_PROJECT", "WANDB_API_KEY"])
         run = None
 
+    sagemaker_config = parse_sm_config()
+    if sagemaker_config:
+        # Set run_id and potentially grouping if we're in SageMaker
+        os.environ['WANDB_RUN_ID'] = '-'.join([
```

Comment:
If TRAINING_JOB_NAME and CURRENT_HOST are not available we'll always use the same run ID which doesn't seem great.